compatibility_server/pip_checker.py

```diff
@@ -39,11 +39,10 @@
 CONTAINER_WITH_PKG = "checker"
 TIME_OUT = 300  # seconds
 
-# Pattern for pip installation errors not related to the package being
-# installed. See:
-# https://github.com/pypa/pip/blob/3a77bd667cc68935040563e1351604c461ce5333/src/pip/_internal/commands/install.py#L533
-PIP_ENVIRONMENT_ERROR_PATTERN = re.compile(
-    r'not install packages due to an EnvironmentError: (?P<error>.*)')
+# Pattern for pip installation error related to the package being
+# installed.
+PIP_INSTALL_ERROR_PATTERN = re.compile(
```

Comment:
If I understand correctly, the idea is that the only time that pip will return a non-zero error code that is package-related is when the package doesn't exist. And you think that is cleaner than the previous approach. Could you document that and maybe point to some source that shows that is true?

Reply:
Yeah I was thinking that if there is a docker timeout error during the pip install process, it will also be treated as an `INSTALL_ERROR`  in the result, which actually isn't. And there might still be some other type of error happened during the installation, probably we could restrict that install error only means the package and version doesn't exist.

Reply:
A docker timeout should lead to a `ReadTimeout`, which results in
```raise PipCheckerError(...)```

Are you sure that a non-zero `pip install` return can only be one of:
1. the package/version not existing
2. some sort of internal error